### PR TITLE
VDS-2474: Add password validation query

### PR DESCRIPTION
**Pull request by pushnitsa**
## Description
Add query for password validation

image

### CODE REVIEW

**Review by asvishnyakov — APPROVED** [review]

File: src/VirtoCommerce.ExperienceApiModule.Web/Module.cs [hunk 1]
```diff
@@ -61,6 +64,7 @@ public void Initialize(IServiceCollection services)
 
 
             services.AddSingleton<IStoreCurrencyResolver, StoreCurrencyResolver>();
+            services.TryAddSingleton<Func<IPasswordValidator<ApplicationUser>>>(provider => () => provider.CreateScope().ServiceProvider.GetService<IPasswordValidator<ApplicationUser>>());
```
Comment (Vectorfield4) [inline src/VirtoCommerce.ExperienceApiModule.Web/Module.cs:67]: Why do we use factory instead of direct DI?

Comment (pushnitsa) [reply src/VirtoCommerce.ExperienceApiModule.Web/Module.cs:67]: Cause it is scoped service, and I need factory, which is missed

Comment (Vectorfield4) [reply src/VirtoCommerce.ExperienceApiModule.Web/Module.cs:67]: `UserManager` already has `PasswordValidators` insde.

Comment (Vectorfield4) [reply src/VirtoCommerce.ExperienceApiModule.Web/Module.cs:67]: https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.identity.usermanager-1?view=aspnetcore-5.0#properties

Comment (Vectorfield4) [reply src/VirtoCommerce.ExperienceApiModule.Web/Module.cs:67]: So, can you use https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.identity.usermanager-1.validatepasswordasync?view=aspnetcore-5.0#Microsoft_AspNetCore_Identity_UserManager_1_ValidatePasswordAsync__0_System_String_ ?

File: src/VirtoCommerce.ExperienceApiModule.Core/Queries/PasswordValidationQueryHandler.cs [hunk 2]
```diff
@@ -11,29 +12,36 @@ namespace VirtoCommerce.ExperienceApiModule.Core.Queries
...
+
+            foreach (var passwordValidator in userManager.PasswordValidators)
+            {
+                var validationResult = await passwordValidator.ValidateAsync(userManager, null, request.Password);
+
+                // If at least one validator fails, result should be failed
+                if (!validationResult.Succeeded)
+                {
+                    result.Succeeded = validationResult.Succeeded;
+                }
```
Comment (Vectorfield4) [inline src/VirtoCommerce.ExperienceApiModule.Core/Queries/PasswordValidationQueryHandler.cs:39]:
```
result.Succeeded &= validationResult.Succeeded;
```

**Review by Vectorfield4 — APPROVED** [review]

**Pull request merged**